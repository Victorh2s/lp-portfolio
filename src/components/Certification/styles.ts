import styled, { css } from 'styled-components';

export const Certification = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.large};
    text-decoration: none;
    font-family: ${theme.font.family.main};
    margin-top: 0.6rem;


  @media ${theme.media.lteMedium}{
    > a{
      font-size: 1.4rem;
    }
  }

  `}
`;

export const a = styled.a`
  ${({ theme }) => css`

    >svg {
    width: 2rem;
    height: 2rem;
    color: ${theme.colors.green};
  }



  `}
`;
