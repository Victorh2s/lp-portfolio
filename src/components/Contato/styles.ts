import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  align-items: center;
  padding: ${theme.spacings.xlarge};

  `}
`;

export const Nav = styled.nav`
display: flex;
  align-items: center;
  justify-content: center;
`;

export const Icons = styled.div`

`;

export const Container = styled.a`
  ${({ theme }) => css`

  color:${theme.colors.white};
  padding: ${theme.spacings.small};

  >svg{
    width: 5rem;
    height: 5rem;


  }

  @media ${theme.media.lteMedium}{
    >svg{
    width: 3rem;
    height: 3rem;
  }
  }

  `}
`;
