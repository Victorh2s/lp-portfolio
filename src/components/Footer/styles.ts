import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    background: ${theme.colors.dark};
    border-top: 0.1rem solid ${theme.colors.green};
    align-items: center;
    justify-content: center;
    padding: 1rem;
  `}
`;

export const Container = styled.a`
    border: 0.1rem solid;
    border-radius: 50rem;
    width: 4rem;
    height: 4rem;
    z-index: 6;
`;
