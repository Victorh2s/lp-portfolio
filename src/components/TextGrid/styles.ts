import styled, { css } from 'styled-components';
import { TextGridProps } from '.';

export const Wrapper = styled.p<TextGridProps>`
  ${({ theme, colorDark }) => css`

  color: ${colorDark ? theme.colors.dark : theme.colors.white};
  font-size: ${theme.font.sizes.small};
  font-family:${theme.font.family.main};
  padding: 0;

  `}
`;
