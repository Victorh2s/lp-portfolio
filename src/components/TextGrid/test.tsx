import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { TextGrid, TextGridProps } from '.';
import { theme } from '../../styles/theme';

const props: TextGridProps = {
  children: 'testando',
};

describe('<TextGrid />', () => {
  it('should render TextGrid in the document', () => {
    renderTheme(<TextGrid {...props} />);

    expect(screen.getByText('testando')).toBeInTheDocument();
  });

  it('should render TextGrid with ColorDark false', () => {
    renderTheme(<TextGrid colorDark={false} {...props} />);

    expect(screen.getByText('testando')).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render TextGrid with ColorDark true', () => {
    renderTheme(<TextGrid colorDark={true} {...props} />);

    expect(screen.getByText('testando')).toHaveStyle({
      color: theme.colors.dark,
    });
  });

  it('should match to snapshot', () => {
    const { container } = renderTheme(<TextGrid {...props} />);

    expect(container).toMatchSnapshot();
  });
});
