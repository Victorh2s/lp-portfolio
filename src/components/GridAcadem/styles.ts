import { Wrapper as TextGrid } from '../TextGrid/styles';
import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Certification } from '../Certification/styles';

export const Grid = styled.main`
  ${({ theme }) => css`

  @media ${theme.media.lteMedium}{
    ${Heading}{
      font-size: ${theme.font.sizes.medium};
    }


}
  `}
`;

export const Container = styled.section`
  ${({ theme }) => css`

  ${TextGrid}{
    margin-top: -0.1rem;
  }

  @media ${theme.media.lteMedium}{
    ${TextGrid}{
    font-size: 1.4rem;
  }


}

  `}
`;

export const Section = styled.div`
  ${({ theme }) => css`
  display: flex;


  @media ${theme.media.lteMedium}{
      display: block;
      align-items:center;
      ${Certification}{
        margin-top: -0.5rem;
        margin-bottom: 1rem;
        padding: 0;
      }

  }

  `}
`;

export const Title = styled.div`
  ${({ theme }) => css`

  display: flex;
  align-items: center;
  ${Heading}{
    padding: 1rem;
    }



  @media ${theme.media.lteMedium}{
    ${Image}{
      width: 5rem;
      height: 4rem;
      padding: 0.1rem;
    }
    ${Heading}{
      font-size: 2rem;
      padding: 0;
    }

    display: inline;
  }


  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
  width: 6rem;
  height: 5rem;
  padding: 0.3rem;
  border-radius: 10%;
  background: ${theme.colors.white};

  `}
`;
