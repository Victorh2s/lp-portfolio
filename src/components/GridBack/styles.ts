import { Title as Heading } from '../Heading/styles';
import { Wrapper as TextGrid } from '../TextGrid/styles';
import styled, { css } from 'styled-components';

export const Front = styled.div`
  ${({ theme }) => css`
  padding: ${theme.spacings.xlarge};

${Heading}{
    text-align: center;
}

border: 0.1rem solid;
border-image-slice: 1;
border-width: 0.1rem;
border-image-source: linear-gradient(to right, #090213, #078a85);

@media ${theme.media.lteMedium}{
padding: ${theme.spacings.small};
}

  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(2, 2fr);
    gap: ${theme.spacings.small};



    @media ${theme.media.lteMedium}{
      display: block;
      ${TextGrid}{
        font-size: ${theme.font.sizes.small};
      }
      ${Image}{
        width: 7rem;
        height: 7rem;
      }
  }
  `}
`;

export const container = styled.div`
  ${({ theme }) => css`
text-align: center;
justify-content: center;

>svg{
  color:${theme.colors.white};
  width:10rem;
  height:10rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
  transition: all 200ms ease-in-out;
  padding: ${theme.spacings.small};
  cursor: pointer;

  &:hover{
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.4rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
  }
}

${TextGrid}{
  margin-top: 0.8rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to right, #090213, #078a85);
}
`}
`;

export const Image = styled.img`
  width:10rem;
  height:10rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
  transition: all 200ms ease-in-out;

  cursor: pointer;
  &:hover{
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.5rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
  }
`;
