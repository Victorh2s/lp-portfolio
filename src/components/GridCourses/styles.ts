import { Wrapper as TextGrid } from '../TextGrid/styles';
import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Certification } from '../Certification/styles';

export const Grid = styled.main`
  ${({ theme }) => css`

  @media ${theme.media.lteMedium}{
    ${Heading}{
      font-size: ${theme.font.sizes.medium};
    }

}
  `}
`;

export const Container = styled.section`
  ${({ theme }) => css`
  border-bottom: 0.1rem solid ${theme.colors.green};

  ${TextGrid}{
    margin-top: -0.1rem;
  }

  @media ${theme.media.lteMedium}{
    ${TextGrid}{
    font-size: 1.4rem;
  }

}

  `}
`;

export const Section = styled.div`
  ${({ theme }) => css`
  display: flex;


  @media ${theme.media.lteMedium}{
      display: block;
      align-items:center;
      ${Certification}{
        margin-top: -0.5rem;
        margin-bottom: 1rem;
        padding: 0;
      }

  }

  `}
`;

export const Title = styled.div`
  ${({ theme }) => css`

  display: flex;
  align-items: center;

  >svg {
    width: 5rem;
    height: 5rem;
    color: ${theme.colors.white};
  }

  @media ${theme.media.lteMedium}{
    ${Heading}{
      font-size: 2rem;
    }

}


  `}
`;
