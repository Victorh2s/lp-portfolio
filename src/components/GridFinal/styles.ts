import styled, { css } from 'styled-components';
import { Wrapper as TextGrid } from '../TextGrid/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`

  display: flex;
  padding: ${theme.spacings.xlarge};
  align-items: center;
  justify-content: center;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to top, #078a85, #090213);


  @media ${theme.media.lteMedium}{
      display: block;
      align-items: center;

      ${Image}{
        width: 7rem;
        height: 7rem;


      }
  }
  `}
`;

export const container = styled.div`
  ${({ theme }) => css`
  padding: ${theme.spacings.xlarge};
  text-align: center;
  justify-content: center;

  ${TextGrid}{
  margin-bottom: -0.1rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to bottom, #090213, #078a85);
}

>svg{
  color:${theme.colors.white};
  width:10rem;
  height:10rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to bottom, #090213, #078a85);
  transition: all 200ms ease-in-out;
  padding: ${theme.spacings.small};
  cursor: pointer;

  &:hover{
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.4rem;
  border-image-source: linear-gradient(to top, #090213, #078a85);
  }
}

@media ${theme.media.lteSmall}{
    ${TextGrid}{
    border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.1rem;
  border-image-source: linear-gradient(to left, #090213, #078a85);
}
  }


  `}
`;

export const Image = styled.img`
  width:10rem;
  height:10rem;
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.2rem;
  border-image-source: linear-gradient(to bottom, #090213, #078a85);
  transition: all 200ms ease-in-out;
  cursor: pointer;

  &:hover{
  border: 0.1rem solid;
  border-image-slice: 1;
  border-width: 0.5rem;
  border-image-source: linear-gradient(to top, #090213, #078a85);
  }
`;
