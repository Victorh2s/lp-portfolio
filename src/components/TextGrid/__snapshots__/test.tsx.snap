// Vitest Snapshot v1

exports[`<TextGrid /> > should match to snapshot 1`] = `
.c0 {
  color: #FFFFFF;
  font-size: 1.6rem;
  font-family: 'Work Sans',sans-serif;
  padding: 0;
}

<div>
  <p
    class="c0"
  >
    testando
  </p>
</div>
`;
